src/wasm/wasm-binary.cpp

```diff
@@ -636,8 +636,13 @@ void WasmBinaryWriter::visitSetLocal(SetLocal *curr) {
 }
 
 void WasmBinaryWriter::visitGetGlobal(GetGlobal *curr) {
-  if (debug) std::cerr << "zz node: GetGlobal " << (o.size() + 1) << std::endl;
-  o << int8_t(BinaryConsts::GetGlobal) << U32LEB(getGlobalIndex(curr->name));
+  if (curr->name.is()) {
```

Comment:
In the binary format code we can assume that the module is valid, so `curr->name` must be a valid value. What are you trying to achieve here, there might be a better way?